rust/operator-binary/src/airflow_controller.rs

```diff
@@ -965,7 +1002,31 @@ fn build_server_rolegroup_statefulset(
         };
         airflow_container.readiness_probe(probe.clone());
         airflow_container.liveness_probe(probe);
-        airflow_container.add_container_port("http", resolved_port.into());
+        airflow_container.add_container_port(HTTP_PORT_NAME, http_port.into());
+    }
+
+    let mut pvcs: Option<Vec<PersistentVolumeClaim>> = None;
+
+    if airflow
+        .merged_listener_class(airflow_role, &rolegroup_ref.role_group)
+        .is_some()
+    {
+        // Listener endpoints for the Webserver role will use persistent volumes
+        // so that load balancers can hard-code the target addresses. This will
+        // be the case even when no class is set (and the value defaults to
+        // cluster-internal) as the address should still be consistent.
+        let pvc = ListenerOperatorVolumeSourceBuilder::new(
+            &ListenerReference::ListenerName(airflow.group_listener_name(rolegroup_ref)),
```

Comment:
Nit: Maybe we should define `group_listener_name` to return an `Option` instead, and `if let` on that, rather require the consuming code (here) to make the implication between `merged_listener_class` and `group_listener_name`?

Reply:
We could eliminate the extra nesting by doing something like `if let (Some(foo), Some(bar)) = (build_foo(), build_bar()) {}`, but I agree that it's weird that they end up seeming uncorrelated there...

Reply:
To be considered in https://github.com/stackabletech/airflow-operator/issues/622